niceman/resource/base.py

```diff
@@ -134,10 +134,12 @@ def factory(config):
         except Exception as exc:
             # Typically it should be an ImportError, but let's catch and recast
             # anything just in case.
+            msg = exc_str(exc)
+            known = ResourceManager._discover_types()
```

Comment:
I wouldn't be surprised if we manage to raise exception there as well, so may be try/except these 3 lines as well and add caught exception message (msg += " Failed to discover resource types: %s" % exc_str(exc2)) in except for it?